File: go/trajectory1.go
package main

import (
	"fmt"
	"math/big"
)

func main() {
	x, _ := new(big.Int).SetString("99123456789", 10)
	zero := big.NewInt(0)
	one := big.NewInt(1)
	two := big.NewInt(2)
	three := big.NewInt(3)
	T := 350

	for t := 0; t < T; t++ {
		div := new(big.Int)
		if div.Mod(x, two).Cmp(zero) == 0 {
			x = x.Div(x, two)
		} else {
			x = x.Mul(x, three)
			x = x.Add(x, one)
		}

		fmt.Println(x)
	}
}

File: java/src/Trajectory1.java
import java.math.BigInteger;

public class Trajectory1 {
    public static void main(String[] args) {
        BigInteger X = new BigInteger("998123456789");
        BigInteger TWO = new BigInteger("2");
        BigInteger THREE = new BigInteger("3");
        int T = 350;
        for (int t=0; t < T; t++) {
            if (X.mod(TWO).equals(BigInteger.ZERO)) X = X.divide(TWO);
            else X = X.multiply(THREE).add(BigInteger.ONE);
            System.out.println(X);
        }
    }
}
